### PR TITLE
created mast state and added a pause before landing

### LINKED ISSUE #207: Fly to Mast
## Description of Problem

Given a predetermined location fly the drone to the mast

### PR DESCRIPTION
**Pull request by noahklein00**
Added a new state to the __init__.py file called "mast". After completing "early_laps" state, this sends the drone to the "mast" location from the config.py file, sleeps for 20 seconds, then lands.

### CODE REVIEW

File: flight/states/mast.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import asyncio
+import math
+import mavsdk as sdk
+
+from flight import config
+
+from flight.utils.movement_controller import MovementController
+from .land import Land
+
+class Mast:
```
Comment (ewad3) [inline flight/states/mast.py]: The class mast should be a child class of state, this also requires importing state. ex: class Mast(State):

File: flight/states/early_laps.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 from flight.utils.movement_controller import MovementController
 from .land import Land
```
Comment (ewad3) [inline flight/states/early_laps.py]: Remove Land if it is not in use.

File: flight/states/early_laps.py [hunk 3]
```diff
@@ -39,4 +40,5 @@ async def run(self, drone):
 
             logging.debug("Lap %d: Turn two", i)
             await mover.turn(drone)  # turn around pylon 1
-        return Land()
+        #return Land()
```
Comment (ewad3) [inline flight/states/early_laps.py]: Remove Land if it is not in use.

File: flight/config.py [hunk 4]
```diff
@@ -33,16 +33,23 @@
 lon2: Longitude = Longitude(degree=-91, minute=-47, second=0)
 pylon2: LatLon = LatLon(lat2, lon2)
 
+# Position for the mast
+mast1: Latitude = Latitude(degree=37, minute=56, second=53.0) # placeholder postion
+mast2: Longitude = Longitude(degree=-91, minute=-47, second=-5.0)
```
Comment (ewad3) [inline flight/config.py]: The naming of these don't make much sense.

File: flight/config.py [hunk 5]
```diff
@@ -4,7 +4,7 @@
 from mavsdk import System
 
 
-MAX_SPEED: float = 6.352  # m/s
+MAX_SPEED: float = 15  # m/s
```
Comment (ewad3) [inline flight/config.py]: Understand increasing speed for testing but for merging leave as default.

File: flight/config.py [hunk 6]
```diff
@@ -33,16 +33,23 @@
...
+# Position for the mast
+mast1: Latitude = Latitude(degree=37, minute=56, second=53.0) # placeholder postion
+mast2: Longitude = Longitude(degree=-91, minute=-47, second=-5.0)
+mast: LatLon = LatLon(mast1, mast2)
+
 OFFSET: float = 0.005  # km
 DEG_OFFSET: int = 90  # deg
 
-NUM_LAPS: int = 2
+NUM_LAPS: int = 1
```
Comment (ewad3) [inline flight/config.py]: Same as the speed leave as default for merging.

File: flight/config.py [hunk 7]
```diff
@@ -33,16 +33,23 @@
...
 
 THINK_FOR_S: float = 2.0
 FAST_THINK_S: float = 1.0
 
 
 async def config_params(drone: System):
+    await drone.action.set_maximum_speed(MAX_SPEED)
+    return
```
Comment (ewad3) [inline flight/config.py]: Leave default for merging.

File: flight/states/mast.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    """ Sends the drone from the first pylon to the mast """
+    async def run(self, drone):
+        """Sends the drone from the first pylon to the mast"""
+        mover: MovementController = MovementController()
+        # Go to the mast
+        logging.info("Moving to mast")
+        await mover.move_to(drone, config.mast)
```
Comment (ewad3) [inline flight/states/mast.py]: Note about move_to function it fly's to an offset point to the right and not to the intended point. Not your fault but something to consider. May want to look into setting offset based off of a boolean.

**Review by ewad3 — CHANGES_REQUESTED** [review]
Over all looks good just a few things to change and consider.

**Review by ewad3 — DISMISSED** [review]
Looks ready for a flight test.

**Review by pieperm — APPROVED** [review]

**Pull request merged**